src/session.rs

```diff
@@ -1135,19 +1135,31 @@ impl Session {
         for path in paths {
             let path = path.as_ref();
 
+            // by this point the path should be canonicalized, so the path
+            // should not end with a period character
+            if path.to_str().unwrap().ends_with(".") {
+                return Err(io::Error::new(
+                    io::ErrorKind::InvalidInput,
+                    format!("`{:?}` is not a valid filename", path),
+                ));
+            }
```

Comment:
This seems like a very specific thing to check for - can we not let it fail at `path.exists()`?

Reply:
The `if !path.exists() && path.with_extension("png").exists()` check, right?

Specifically the `e` edit command heeds the below logic:

* if a file does not exist...
    * and it does not feature a `.png` extension...
        * create a new file of that name, with an appended `.png` extension as part of the filename

A bug occurs if the end user does the following:

* the user runs a command like `e ~/abc` ...
    * rx sees that `~/abc` does not exist...
        * rx creates a new blank view, which will save to `~/abc.png` even if `~/abc.png` already exists

So in my mind we ought to add a check for this situation to prevent the user from accidentally saving over existing files.

Reply:
The `if !path.exists() && path.with_extension("png").exists()` check, right?

Specifically the `e` edit command heeds the below logic:

* if a file does not exist...
    * and it does not feature a `.png` extension...
        * create a new file of that name, with an appended `.png` extension as part of the filename

A bug occurs if the end user does the following:

* the user runs a command like `e ~/abc` ...
    * rx sees that `~/abc` does not exist...
        * rx creates a new blank view, which will save to `~/abc.png` even if `~/abc.png` already exists

So in my mind we ought to add a check for this situation to prevent the user from accidentally saving over existing files.

Reply:
Or do you mean the `if path.to_str().unwrap().ends_with(".")` check?

That check was to avoid accidently opening files like `/tmp/a..`, though I suppose I could remove that check if you want?

Reply:
I mean the later, the one right above my comment. Right, I'm wondering why we need to handle that case differently - I'd expect that removing the check yields an error anyway.

Reply:
I have since removed it.